Solution

AVyha opened the pull request.

AVyha committed:
Solution

AVyha committed:
update

AVyha committed:
flake8 problem fixed

AVyha committed:
flake8 fixed

mltdll reviewed (changes requested):
Wow. You implemented everything in a completely different way than the instructions suggested. I'm not sure if this means the task has to be rejected.

AVyha commented on app/main.py:
```diff
@@ -1,34 +1,68 @@
+            self.field[deck.row][deck.column] = "x"
+
+
+class Battleship(Ship):
```
If i remove this inherit i should change 3 methods in Ship( Maybe later i try to change but now i dont have idea how to easy rebuild this block :)

Nattalli replied:
that is the truth, you may to fix it

mltdll reviewed (approved):
I guess it works...

AVyha commented:
> I guess it works...

i think you are right that i need remove inherit, im just say that im very lazy to do it now xD

AVyha committed:
Solution2 with extra

solution with extra task but with bad solution (not my opinion) im gonna try to rebuild it but still hope that you aprove it :)

Nattalli commented on app/main.py:
```diff
@@ -1,34 +1,176 @@
+        column: int
+    ) -> Deck | Ship:
+
+        for ship in self.ships:
```
I see only `self.ship` in `__init__()` method. Where the `self.ships` is created?

Nattalli commented on app/main.py:
```diff
@@ -1,34 +1,176 @@
+                                    "position away from the other ship"
+                                )
+
+    def check_all_ships_on_field(self) -> None:
+        must_have = {
+            1: 4,
+            2: 3,
+            3: 2,
+            4: 1
+        }
+        we_have = {
+            1: 0,
+            2: 0,
+            3: 0,
+            4: 0
+        }
+        for ship in self.ships:
+            we_have[ship.len_ship] += 1
+
+        assert must_have == we_have, \
+            "We need 1 four-deck ship, 2 three-deck ships, " \
+            "3 double-deck ships and 4 single-deck ships"
```
did you add your own test? Or why do you need it? That is a hard-code I think

Nattalli commented on app/main.py:
```diff
@@ -1,34 +1,176 @@
+            "3 double-deck ships and 4 single-deck ships"
+
+
+if __name__ == "__main__":
+    battle_ship = Battleship(
+        ships=[
+            ((8, 3), (8, 6)),
+            ((4, 3), (6, 3)),
+            ((4, 5), (6, 5)),
+            ((4, 1), (5, 1)),
+            ((2, 2), (2, 3)),
+            ((3, 7), (4, 7)),
+            ((2, 0), (2, 0)),
+            ((0, 1), (0, 1)),
+            ((0, 4), (0, 4)),
+            ((1, 6), (1, 6)),
+        ]
+    )
+
+    # print(
+    #     battle_ship.fire((9, 7)), #miss
+    #     battle_ship.fire((7, 9)), #miss
+    #     battle_ship.fire((9, 9)), #miss
+    #     battle_ship.fire((2, 8)), #miss
+    #     battle_ship.fire((2, 9)), #miss
+    # )
+    # print(
+    #     battle_ship.fire((8, 3)),  #hit
+    #     battle_ship.fire((8, 4)),  #hit
+    #     battle_ship.fire((8, 5)),  #hit
+    #     battle_ship.fire((8, 6)),  #sunk
+    #     battle_ship.fire((0, 1)),  #sunk
+    #     battle_ship.fire((9, 9)),  #miss
+    #     battle_ship.fire((4, 1))  #hit
+    # )
+
```
also you can remove it from commit

AVyha committed:
inheritance removed

Nattalli reviewed (approved):
Excellent! 🌞
